src/engine/storage.h

```diff
@@ -21,15 +21,17 @@ class IStorage : public IInterface
 
 	virtual void ListDirectory(int Type, const char *pPath, FS_LISTDIR_CALLBACK pfnCallback, void *pUser) = 0;
 	virtual IOHANDLE OpenFile(const char *pFilename, int Flags, int Type, char *pBuffer = 0, int BufferSize = 0) = 0;
-	virtual bool FindFile(const char *pFilename, const char *pPath, int Type, char *pBuffer, int BufferSize, unsigned WantedCrc = 0, unsigned WantedSize = 0) = 0;
+	virtual bool FindFile(const char *pFilename, const char *pPath, int Type, char *pBuffer, int BufferSize) = 0;
+	virtual bool FindFile(const char *pFilename, const char *pPath, int Type, char *pBuffer, int BufferSize, unsigned WantedCrc, unsigned WantedSize) = 0;
```

Comment:
Question: what is the use of splitting this function in two instead of using default values for the arguments? It seems like `FindFile(..., BufferSize)`  does the same as `FindFile(..., BufferSize, 0, 0)`. Am I mistaken?

Reply:
It allows us to distinguish between the case that the user wants a file with CRC 0 and size 0 and the case where the user doesn't care about CRC and size.

The calls are not equivalent, in the first case we don't check for CRC and size, in the second case we make sure both are 0.

Reply:
> The calls are not equivalent, in the first case we don't check for CRC and size, in the second case we make sure both are 0.

Sorry, I don't see where you make sure both are 0 in the second case. To me, it would be completely equivalent to keep

```cpp
virtual bool FindFile(const char *pFilename, const char *pPath, int Type, char *pBuffer, int BufferSize, unsigned WantedCrc = 0, unsigned WantedSize = 0) = 0;```

only

Reply:
`Data.m_CheckCrcSize` is set to `true` or `false` depending on which of these functions is called. CRC and size are only checked when this variable is set.

Reply:
Ah, I missed that, thanks :)